[2025-08-14T21:47:15.839+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T03:48:52+00:00 [queued]>
[2025-08-14T21:47:15.844+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T03:48:52+00:00 [queued]>
[2025-08-14T21:47:15.845+0000] {taskinstance.py:2170} INFO - Starting attempt 12 of 12
[2025-08-14T21:47:15.877+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): fetch_historical_data> on 2025-08-14 03:48:52+00:00
[2025-08-14T21:47:15.886+0000] {standard_task_runner.py:60} INFO - Started process 31646 to run task
[2025-08-14T21:47:15.893+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'stock_market_batch_pipeline', 'fetch_historical_data', 'manual__2025-08-14T03:48:52+00:00', '--job-id', '40', '--raw', '--subdir', 'DAGS_FOLDER/stock_market_batch_dag.py', '--cfg-path', '/tmp/tmp3n25ban_']
[2025-08-14T21:47:15.903+0000] {standard_task_runner.py:88} INFO - Job 40: Subtask fetch_historical_data
[2025-08-14T21:47:15.977+0000] {task_command.py:423} INFO - Running <TaskInstance: stock_market_batch_pipeline.fetch_historical_data manual__2025-08-14T03:48:52+00:00 [running]> on host 0181ae2045ff
[2025-08-14T21:47:16.035+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Basanth Kumar' AIRFLOW_CTX_DAG_ID='stock_market_batch_pipeline' AIRFLOW_CTX_TASK_ID='fetch_historical_data' AIRFLOW_CTX_EXECUTION_DATE='2025-08-14T03:48:52+00:00' AIRFLOW_CTX_TRY_NUMBER='12' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-14T03:48:52+00:00'
[2025-08-14T21:47:16.036+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-08-14T21:47:16.037+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\n    unset PYTHONNOUSERSITE\n    python /opt/***/dags/scripts/batch_data_producer.py 2025-08-14\n    ']
[2025-08-14T21:47:16.050+0000] {subprocess.py:86} INFO - Output:
[2025-08-14T21:47:16.845+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-08-14T21:47:16.855+0000] {subprocess.py:93} INFO -   File "/opt/***/dags/scripts/batch_data_producer.py", line 8, in <module>
[2025-08-14T21:47:16.856+0000] {subprocess.py:93} INFO -     import yfinance as yf
[2025-08-14T21:47:16.857+0000] {subprocess.py:93} INFO - ModuleNotFoundError: No module named 'yfinance'
[2025-08-14T21:47:16.937+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-08-14T21:47:16.994+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/bash.py", line 212, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-08-14T21:47:17.009+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=stock_market_batch_pipeline, task_id=fetch_historical_data, execution_date=20250814T034852, start_date=20250814T214715, end_date=20250814T214717
[2025-08-14T21:47:17.052+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 40 for task fetch_historical_data (Bash command failed. The command returned a non-zero exit code 1.; 31646)
[2025-08-14T21:47:17.092+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-08-14T21:47:17.188+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
